runtime/hsa-runtime/core/inc/amd_aie_aql_queue.h

```diff
@@ -100,7 +128,7 @@ class AieAqlQueue : public core::Queue,
                        void *value) override;
 
   // AIE-specific API
-  AieAgent &GetAgent() { return agent_; }
+  AieAgent &GetAgent() const { return agent_; }
```

Comment:
If you're giving non-const access to `agent_` then this function should be non-const.

Reply:
Good catch.